tiledb/sm/query/reader.cc

```diff
@@ -2499,6 +2502,13 @@ Status Reader::init_read_state() {
   RETURN_NOT_OK(
       config.get<uint64_t>("sm.memory_budget_var", &memory_budget_var, &found));
   assert(found);
+  offsets_format_ = config.get("sm.offsets_format", &found);
+  if (offsets_format_ != "bytes" && offsets_format_ != "elements") {
+    return LOG_STATUS(
+        Status::ReaderError("Cannot initialize reader; Unsupported offsets "
+                            "format in configuration"));
+  }
+  assert(found);
```

Comment:
move this `assert` immediately after L2505 so that we assert `offsets_format_` is valid before we use it.